src/main/resources/lib/listeners.ts

```diff
@@ -0,0 +1,45 @@
+import { Content, QueryResponse } from 'enonic-types/lib/content'
+import { HighchartConfig } from '../site/macros/highchart/highchart-config'
+import { KeyFigure } from '../site/content-types/keyFigure/keyFigure'
+import { EnonicEvent } from 'enonic-types/lib/event'
+import {DatasetRepoNode} from "./repo/dataset";
+import {JSONstat} from "./types/jsonstat-toolkit";
+import {TbmlData} from "./types/xmlParser";
+import {CreateOrUpdateStatus} from "./ssb/dataset/dataset";
+
+const {
+  getDataset
+} = __non_webpack_require__( './ssb/dataset/dataset')
+const {
+  listener
+} = __non_webpack_require__('/lib/xp/event')
+const {
+  query
+} = __non_webpack_require__('/lib/xp/content')
+const {
+  DataSource
+} = __non_webpack_require__('/lib/repo/dataset')
+const {
+  refreshDataset
+} = __non_webpack_require__( '/lib/ssb/dataset/dataset')
+
+
+export function setupFetchDataOnCreateListener() {
+  listener({
+    type: 'node.updated',
+    callback: function(event: EnonicEvent) {
+      const contentWithDataSource: QueryResponse<HighchartConfig | KeyFigure> = query({
+        query: `_id = '${event.data.nodes[0].id}' AND 
+          ( data.dataSource._selected = '${DataSource.STATBANK_API}' OR data.dataSource._selected = '${DataSource.TBPROCESSOR}' )`,
+        contentTypes: [`${app.name}:highchart`, `${app.name}:keyFigure`],
+        filters: {
+          exists: {
+            field: `data.dataSource.*.urlOrId`
+          }
+        }
+      })
+      contentWithDataSource.hits.forEach((content: Content<HighchartConfig | KeyFigure>) =>  refreshDataset(content))
```

Comment:
not a candidate for async fetch ?

Reply:
Not really :) It will always be one (probably).